src/store/index.js

```diff
@@ -329,8 +331,25 @@ function getStore(config) {
         }
       },
       queryParameters(state, params) {
-        for(let key in params) {
-          state[`${key}QueryParameters`] = params[key];
+        const appState = state.stateQueryParameters;
+        for (let [key, value] of Object.entries(params.state)) {
+          if (Array.isArray(appState[key]) && !(Array.isArray(value))) {
+            value = value.split(',');
+          }
+          Vue.set(appState, key, value);
+        }
+      },
+      addUidToOpenAssets (state, uid) {
+        const idx = state.stateQueryParameters.assets.indexOf(uid);
+        // need to revent duplicates because of the way the collapse v-model works
```

Comment:
So would it make sense to use a Set instead of an array?

Reply:
That would be an option but would probably then require additional code in the `StacBrowser.vue` so it's probably six of one and half a dozen of the other...

Reply:
Fair for the moment, but I assume that we add more places where we add state in such a way and then it's likely easier to just handle Sets once in STACBrowser.vue instead of doing the duplicate check in each component. So for the long-term I think the change would make sense.